packages/nimble-components/src/table/specs/table-column-specs/table-column-anchor-hld.md

```diff
@@ -0,0 +1,141 @@
+# Anchor Table Column HLD
+
+## Problem Statement
+
+We need to support links in table cells. We will do this by creating a new column type.
+
+## Links To Relevant Work Items and Reference Material
+
+[GitHub issue: Table allows configuration of hyperlink column type (#1012)](https://github.com/ni/nimble/issues/1012)
+
+## Implementation / Design
+
+This is a fairly straightforward execution of our custom column type pattern. The anchor column will use two fields from the data records: one for the link's visible text (the "label") and one for the url (the "href"). The column configuration will contain the rest of the native anchor parameters: `hreflang`, `ping`, `referrerpolicy`, `rel`, `target`, `type`, and `download`. The assumption is that these should be the same for each link of a given column (if used at all). An alternative would be to have a `*FieldName` property for each of the native anchor parameters, but that seems unlikely to be useful.
+
+If a table record is missing a href value a text span will be rendered rather than a link.
+
+The column will also take an optional `placeholder` value to use when a record does not define a label.
```

Comment:
My concern with this behavior is that this can ultimately allow behaviors where multiple `hrefs` are provided in the record, with no labels, and they will all appear the same in the table, but navigate to different URLs (which I understand is somewhat mitigated by hover state). If we feel the hover state is enough to overcome the confusion of this unlikely state, I guess I'm fine with that, otherwise I would be interested if @RickA-NI or @NIbokeefe had an opinion on whether we should be showing some invalid configuration state in this scenario.

Reply:
> multiple hrefs are provided in the record, with no labels, and they will all appear the same in the table

We may not be on the same page here. If a record provides a `href` but no `label` for a given anchor column, that cell will display either a blank or the placeholder, but neither will be clickable as a link. So, yes, they would all appear the same in the table (e.g. blank), but hovering won't do anything. Making the label a prerequisite to being a link seems logical to me.

Reply:
If you have an `href` but no label/placeholder, could we just display the URL rather than make it not clickable?

Reply:
What if we have a placeholder but no label? I don't think we should ever turn the placeholder text into a link. Would the URL take precedence over the placeholder? I'm not necessarily opposed to this, but it seems a little more complicated for unclear benefit.

Reply:
What about logic that looks something like the following? It should ensure that the column is always a hyperlink if an href is provided and that it doesn't show a hyperlink without display text associated with it.

```
if (href) {
    if (label) {
        // nimble-anchor with label & href
    } else {
        // nimble-anchor with href only (use the href as the display value)
    }
} else {
    if (label) {
        // show the label as a span
    } else {
        // show the placeholder as a span
    }
}
```

Reply:
That logic will definitely work. I was just saying that the behavior is a little more complicated to describe (and maybe unexpected?), but I'm fine with it if that's what we want.

Reply:
Updated HLD to say we will show the URL if the label is missing.